lib/flatten.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+function flatten(arr, ret) {
+  ret = ret || [];
+  for (var i = 0, n = arr.length; i < n; i++) {
+    var el = arr[i];
+    if (Array.isArray(el)) {
+      flatten(el, ret);
+    } else {
+      ret.push(el)
+    }
+  }
+  return ret;
+}
```

Comment:
When I originally designed bach, I only wanted to support this syntax:
```js
bach.series([a, b, c], options)
```

I don't really care if we recursively flatten, and this will be a breaking release, so I think we can use `Array.prototype.slice.apply`

Reply:
@phated  OK. I'll modify it to use `Array.prototype.slice.apply`.